migrating_gazebo_classic_ros2_packages.md

```diff
@@ -0,0 +1,593 @@
+        '-z', '0.01'
+    ],
+    output='screen',
+)
+```
+
+If you uncomment `ld.add_action(spawn_turtlebot_cmd)` in `empty_world.launch.py`
+and run the launch file, you'll notice errors related to unrecognized plugins.
+These are coming from the model SDFormat file, which we will modify next.
+
+## Modify the model
+
+We will be using the `waffle` robot for this tutorial, so we'll edit the file
+`turtlebot3_gazebo/models/turtlebot3_waffle/model.sdf`. The changes we need to
+make are mostly related to plugins and their parameters.
+
+### libgazebo_ros_imu_sensor.so
+
+This plugin can be removed since there is a generic IMU plugin that handles all
+IMU sensors. We will add this to the world later. We will set the `<topic>` tag
+inside `<sensor>` to a short topic name to make it easier when creating a ROS
+bridge later. The entire `<sensor>` tag should now look like:
+
+```xml
+<link name="imu_link">
+  <sensor name="tb3_imu" type="imu">
+    <always_on>true</always_on>
+    <update_rate>200</update_rate>
+    <topic>imu</topic>
+    <imu>
+      ... <!-- all the content of <imu> -->
+    </imu>
+  </sensor>
+</link>
+```
+
+### libgazebo_ros_ray_sensor.so
+
+Similar to the IMU, we will use a generic plugin loaded into the world for
+handling all rendering sensors, which includes Lidar sensors. Currently the
+`ray` sensor type, which meant to use the physics engine for generator the
+sensor data, is not supported in the new Gazebo, we will need to update it to
+`gpu_lidar`. We'll also need to change the `<ray>` tag inside `<sensor>` to
+`<lidar>`. The `frame_name` parameter of the plugin will be handled by setting
+the `gz_frame_id` parameter in `<sensor>`. Lastly, we'll set the `<topic>`
+parameter similar to the IMU sensor. The final `<sensor>` tag for the Lidar
+should look like:
+
+```xml
+<sensor name="hls_lfcd_lds" type="gpu_lidar">
+  <always_on>true</always_on>
+  <visualize>true</visualize>
+  <pose>-0.064 0 0.121 0 0 0</pose>
+  <update_rate>5</update_rate>
+  <topic>scan</topic>
+  <gz_frame_id>base_scan</gz_frame_id>
+  <lidar>
+    ... <!-- same content as <ray> in the original -->
+  </lidar>
+</sensor>
+```
+
+### libgazebo_ros_camera.so
+
+Remove plugin, set `<topic>` if desired.
+
+- TODO: need to check camera info
+
+### libgazebo_ros_diff_drive.so
+
+Since this is a model specific plugin, we will replace it with the `DiffDrive`
+plugin. We will match the parameters of `libgazebo_ros_diff_drive` as much as
+possible, but exact match may not be possible. For example, the original plugin
+has a `max_wheel_acceleration`, but `gz-sim-diff-drive-system` has
+`max_linear_acceleration` instead, which are not equivalent; the latter is a
+limit on the whole vehicle's linear acceleration. We can approximate the value
+by multiplying the wheel acceleration limit by the radius of the wheel. Refer to
+the
+[DiffDrive class reference](https://gazebosim.org/api/gazebo/6/classignition_1_1gazebo_1_1systems_1_1DiffDrive.html)
```

Comment:
Tutorial is in the harmonic folder 
```suggestion
[DiffDrive class reference](https://gazebosim.org/api/sim/8/classgz_1_1sim_1_1systems_1_1DiffDrive.html)
```

Reply:
The tutorial is actually not in the harmonic folder. I started it there, but moved it before I pushed. I forgot to remove it from the harmonic/index.yaml though. 

Since this tutorial is targeting `humble`, I thought it would be best to link to Fortress documentation, but maybe I should link to both?